### ISSUE TITLE
Performed update on 3.3.0 to 3.4.0 on a nondocker instance, version still shows 3.3.0

### ISSUE THREAD

**Issue by Tebrion74** [tootsuite/mastodon#16289]
This may be a bug or I may have done something wrong.   I followed the update steps to update to 3.4.0 and I had no errors or problems, yet the instance still shows that  V3.3.0 in the about information.  I'm confused because it appeared to do stuff.

**Comment by ClearlyClaire** [comment]
Which commands did you type?
What's the output of `git log --oneline -1`?
Did you restart `mastodon-web` and `mastodon-sidekiq`? (`systemctl reload mastodon-web`, `systemctl restart mastodon-sidekiq`)

**Comment by Tebrion74** [comment]
`git log --oneline -1`

outputs: 
`633d17514 (HEAD, tag: v3.3.0) Bump version to 3.3.0 (#15433)`

I don't know what that means, and I have no idea what happened.  I'm positive I did each step as it is detailed in the upgrade instruction on github for 3.4.0 

Now I'm not sure. I tried to repeat the process and it said there was nothing to do.

**Comment by ClearlyClaire** [comment]
`git fetch && git checkout v3.4.0` before doing the other v3.4.0-specific update steps

**Comment by Tebrion74** [comment]
OKay just did that and here is the output:

`~/live$ git fetch && git checkout v3.4.0

remote: Enumerating objects: 99, done.
remote: Counting objects: 100% (99/99), done.
remote: Compressing objects: 100% (85/85), done.
remote: Total 99 (delta 66), reused 18 (delta 14), pack-reused 0
Unpacking objects: 100% (99/99), done.
From https://github.com/tootsuite/mastodon
   aefc41176..56d79818f  l10n_main  -> origin/l10n_main
   92a9fcf0e..fcdae1007  main       -> origin/main
error: Your local changes to the following files would be overwritten by
checkout:
dist/mastodon-sidekiq.service
dist/mastodon-streaming.service
dist/mastodon-web.service
Please commit your changes or stash them before you switch branches.
Aborting`

I replaced some images in v3.3.0 with customized ones, is that why this is
happening?
oopsie daisy

**Comment by ClearlyClaire** [comment]
hm, those aren't images, those are the default systemd unit files, I'm not sure why you would have modified them

you can check the differences by doing `git diff dist/mastodon-*.service`, and you can reset them by doing `git checkout -- dist/mastodon-*.service` before doing the `git checkout v3.4.0` again

**Comment by Tebrion74** [comment]
I didn't modify any files then. I ONLY replaced a few images.

**Comment by Tebrion74** [comment]
I will try that..   Leave it to me to screw up the simplest thing.

**Comment by Tebrion74** [comment]
Now it's all coming back to me.. I had to make a few hacky tweaks as a few
apps were all competing for the same things and I couldn't afford new
hosts, VM's or another dedicated server. I had some custom local host
addresses and changed the directories to obey my structure.   I have the
diff output saved to review so I can change those things back after the
upgrade..   Thank you for your help..    I have installed so many of these
bloody apps, didn't keep track of everything I did. .    If it breaks after
upgrade I'll start another thread..  peace out!
Robert
